In [1]:
import pandas as pd

%cd ..

/run/media/nazif/2F946E411BA61D49/thesis


In [16]:
mb15 = pd.read_csv("data/processed/mirbase/mirbase15.csv")
mb21 = pd.read_csv("data/processed/mirbase/mirbase21.csv")

print(f"length of mb15: {len(mb15)} \nlength of mb21: {len(mb21)}")
print(f"difference: {len(mb21) - len(mb15)}")

length of mb15: 1100 
length of mb21: 2588
difference: 1488


<span style="color:#548F66">


miRBase 21 (TargetScan) has 2.5x amount of miRNAs than miRBase 15 (CLASH)


</span>


In [15]:
print(mb15.head())
print("\n")
print(mb21.head())

            name                sequence
0     hsa-let-7a  AACUAUACAACCUACUACCUCA
1    hsa-let-7a*   GAAAGACAGUAGAUUGUAUAG
2  hsa-let-7a-2*  GGAAAGCUAGGAGGCUGUACAG
3     hsa-let-7b  AACCACACAACCUACUACCUCA
4    hsa-let-7b*  GGGAAGGCAGUAGGUUGUAUAG


              name                sequence
0    hsa-let-7a-5p  AACUAUACAACCUACUACCUCA
1    hsa-let-7a-3p   GAAAGACAGUAGAUUGUAUAG
2  hsa-let-7a-2-3p  GGAAAGCUAGGAGGCUGUACAG
3    hsa-let-7b-5p  AACCACACAACCUACUACCUCA
4    hsa-let-7b-3p  GGGAAGGCAGUAGGUUGUAUAG


In [17]:
mb15_list = mb15.sequence.tolist()
mb21_list = mb21.sequence.tolist()

common_mirna_count = sum(1 for i in mb15_list if i in mb21_list)

print(common_mirna_count)


1077


<span style="color:#548F66">


1077 of 1100 miRNA sequences found in miRBase 15 (CLASH) are also found in miRBase 21 (TargetScan)


</span>
